build/cmake/DefinePlatformSpecifc.cmake

```diff
@@ -63,14 +78,6 @@ if(MSVC)
     else(WITH_MT)
         set(THRIFT_RUNTIME_POSTFIX "md" CACHE STRING "Set runtime library postfix" FORCE)
     endif(WITH_MT)
-
-    # Disable boost auto linking pragmas - cmake includes the right files
-    add_definitions("-DBOOST_ALL_NO_LIB")
-
-    add_definitions("/MP") # parallel build
-    add_definitions("/W3") # warning level 3
-
-    add_definitions("-DUNICODE -D_UNICODE")
```

Comment:
You removed this line but never added it anywhere else.

Reply:
These lines  were moved to 

> 

  elseif(CMAKE_${lang}_COMPILER_ID STREQUAL "MSVC")
    set(CMAKE_${lang}_FLAGS "${CMAKE_${lang}_FLAGS} /MP") # parallel build
    set(CMAKE_${lang}_FLAGS "${CMAKE_${lang}_FLAGS} /W4") # warning level 4
    set(CMAKE_${lang}_FLAGS "${CMAKE_${lang}_FLAGS} /source-charset:utf-8")
  endif()

in the same cmake file,and I changed W3 to W4.
"-DUNICODE -D_UNICODE" was used to make sure unicode source files could be compiled on MSVC,so I use "/source-charset:utf-8" switch.

Reply:
This may exclude anything before MSVC 2015 from working... but that's all CI is covering.  I'm okay with that.  2013 and earlier can have some pretty strange C++ behavior...

Reply:
> 
> 
> This may exclude anything before MSVC 2015 from working... but that's all CI is covering. I'm okay with that. 2013 and earlier can have some pretty strange C++ behavior...

It seems that vs2013 does not have "/source-charset",so I have to return to macros.